src/async/imap/MCIMAPFetchContentToFileOperation.h

```diff
@@ -0,0 +1,47 @@
+//
+//  IMAPFetchContentToFileOperation.h
+//  mailcore2
+//
+//  Created by Dmitry Isaikin on 2/08/16.
+//  Copyright (c) 2016 MailCore. All rights reserved.
+//
+
+#ifndef MAILCORE_IMAPFETCHCONTENTTOFILEOPERATION_H
+
+#define MAILCORE_IMAPFETCHCONTENTTOFILEOPERATION_H
+
+#include <MailCore/MCIMAPOperation.h>
+
+#ifdef __cplusplus
+
+namespace mailcore {
+
+    class MAILCORE_EXPORT IMAPFetchContentToFileOperation : public IMAPOperation {
+    public:
+        IMAPFetchContentToFileOperation();
+        virtual ~IMAPFetchContentToFileOperation();
+
+        virtual void setUid(uint32_t uid);
+        virtual void setPartID(String * partID);
+        virtual void setEncoding(Encoding encoding);
+        virtual void setOutputFile(String * outputFile);
+        virtual void setChunkSize(uint32_t chunkSize);
```

Comment:
No, currently only one. But I don't know about the future.

It may be worthwhile to remove the information about concrete way of attachment loading from interface?
`MCOIMAPFetchContentToFileOperation * op = [session fetchMessageAttachmentToFileOperationWithFolder:@"INBOX"
                                                                                                    uid:456
                                                                                                 partID:@"1.2"
                                                                                               encoding:MCOEncodingBase64
                                                                                                 toFile:fileNane
                                                                                                 urgent:YES];

     [op setLoadingByChunksWithSize:1024*1024
                              withEstimatedSize:sizeOfAttachFromBodystructure];

     [op start:^(NSError * __nullable error) {
         ...
     }];`

The usual way would be loading the full part as a whole, and setLoadingByChunksWithSize:withEstimatedSize: message would enable alternative chunked loading approach.